addons/l10n_cl/models/account_move.py

```diff
@@ -27,23 +26,92 @@ def _get_l10n_latam_documents_domain(self):
+                    partner_domain += [('code', 'in', ['70', '71', '56', '61'])]
+                elif self.partner_id_vat == '60805000-0':
+                    partner_domain += [('code', '=', '914')]
+                elif self.partner_id.l10n_cl_sii_taxpayer_type == '3':
+                    partner_domain += [('code', 'in', ['35', '38', '39', '41', '56', '61'])]
+                elif not self.partner_id.l10n_cl_sii_taxpayer_type or self.partner_id.country_id != self.env.ref(
+                        'base.cl') or self.partner_id.l10n_cl_sii_taxpayer_type == '4':
+                    partner_domain += [('code', 'in', [])]
+                document_type_obj = self.env['l10n_latam.document.type'].search(partner_domain)
+                document_type_ids = document_type_obj.ids
             domain += [('id', 'in', document_type_ids)]
-            if self.partner_id.l10n_cl_sii_taxpayer_type == '3':
-                domain += [('code', 'in', ['35', '38', '39', '41', '56', '61'])]
         return domain
 
-    @api.constrains('type', 'l10n_latam_document_type_id')
+    @api.depends('journal_id', 'partner_id', 'company_id', 'type')
+    def _compute_l10n_latam_available_document_types(self):
+        for record in self:
+            if record.journal_id.l10n_latam_use_documents and \
+                    record.journal_id.company_id.country_id == self.env.ref('base.cl'):
+                record.l10n_latam_available_document_type_ids = False
+                for rec in self.filtered(lambda x: x.journal_id and x.l10n_latam_use_documents):
+                    rec.l10n_latam_available_document_type_ids = self.env['l10n_latam.document.type'].search(
+                        rec._get_l10n_latam_documents_domain())
+            else:
+                super()._compute_l10n_latam_available_document_types()
+
+    @api.constrains('state', 'type', 'l10n_latam_document_type_id')
     def _check_invoice_type_document_type(self):
         super()._check_invoice_type_document_type()
-        for rec in self.filtered(lambda r: r.company_id.country_id == self.env.ref(
-                'base.cl') and r.l10n_latam_document_type_id):
+        for rec in self.filtered(
+                lambda r: r.company_id.country_id == self.env.ref('base.cl') and r.state in ['posted']):
             tax_payer_type = rec.partner_id.l10n_cl_sii_taxpayer_type
+            vat = rec.partner_id.vat
+            country_id = rec.partner_id.country_id
             latam_document_type_code = rec.l10n_latam_document_type_id.code
-            if not tax_payer_type and latam_document_type_code not in ['35', '38', '39', '41']:
-                raise ValidationError(_('Tax payer type is mandatory for this type of document. '
-                                        'Please set the current tax payer type of this client'))
+            if rec.journal_id.type == 'sale' and rec.journal_id.l10n_latam_use_documents:
+                if not tax_payer_type or not vat:
```

Comment:
Maybe, but actually docs type '35', '38', '39', '41' are only used for sales and not for purchases. When companies use "boletas" for purchases instead of a "facturas" they don't register it as a "boleta"; instead, they create an expenses account entry with a batch of this type of documents, or use the expenses module to surrender them.